### PR TITLE
Make installable for Python 3.10.6

### PR DESCRIPTION
**Pull request by dwoiwode**
Due to [following line](https://github.com/automl/HPOBench/blob/d8b45b1eca9a61c63fe79cdfbe509f77d3f5c779/setup.py#L39), HPOBench is no installable for Python 3.10.6. I use HPOBench in the test cases of my package ([pyPDPPartitioner](https://github.com/dwoiwode/py-pdp-partitioner)) and therefore need to install HPOBench in the test-pipeline. This fails, as this line prohibits the installation.

There are a few workarounds for me:
1. Change `python_requires='>=3.6, <=3.10'` to `python_requires='>=3.6, <3.11'`, which just stalls the issue (Currently done in this branch)
2. Remove upper python boundary so that `python_requires='>=3.6, <=3.10'` will be `python_requires='>=3.6'`
3. Remove HPOBench in my test cases/dependencies

Personally, I do not think there is an issue with the second approach, although this might be untested in future python versions

### CODE REVIEW

**Comment by eddiebergman** [comment]
I think approach 2. is best, there's rarely ever a reason to upper bound a python version

**Comment by PhMueller** [comment]
Hi, 

thanks for your pull request.
Approach 2 is also fine by me.

Could you also please redirect the PR to the development branch?

Thanks

File: setup.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def read_file(file_name):
     version=read_file('hpobench/__version__.py').split()[-1].strip('\''),
     packages=setuptools.find_packages(exclude=['*.tests', '*.tests.*',
                                                'tests.*', 'tests'],),
-    python_requires='>=3.6, <=3.10',
+    python_requires='>=3.6, <3.11',
```
Comment (PhMueller) [inline setup.py]:
```suggestion
    python_requires='>=3.6',
```

**Review by PhMueller — CHANGES_REQUESTED** [review]

**Comment by dwoiwode** [comment]
Looks like the python versions were interpreted as floats instead of strings. I hope the latest commit fixes this

**Comment by PhMueller** [comment]
I think the build for python 3.10 crashes due some old (unnecessary) dependencies that we have in the HPOBench.
We are planing to remove them with the next release (coming end of september). This should fix the error above. 
We also will include the support for py>=3.10 then.

**Comment by eddiebergman** [comment]
We have a similar issue in auto-sklearn, the problem is essentially the scikit learn version. Scikit learn needs an older version of numpy which doesn't release a pre-built wheel for python 3.10. Therefore, it downloads the older version of numpy and tries to build it for python 3.10 and then fails.

**Comment by PhMueller** [comment]
Hey, thanks for your work. I have fixed the remaining error in #155 on our side and included your changes into the dev branch. 

Therefore, i close this PR.

**Pull request closed without merging**